internal/controller/server/repository/job/job.go

```diff
@@ -2,67 +2,87 @@ package job
 
 import (
 	"context"
+	"errors"
+	"fmt"
+	"github.com/golang/protobuf/proto"
+	"octavius/internal/pkg/constant"
 	"octavius/internal/pkg/db/etcd"
+	"octavius/internal/pkg/log"
+	clientCPproto "octavius/internal/pkg/protofiles/client_cp"
+	"octavius/internal/pkg/util"
+	"strconv"
 	"strings"
 )
 
-type JobRepository interface {
-	ExecuteJob(context.Context, string, string, map[string]string) error
-	CheckJobMetadataIsAvailable(context.Context, string) (bool, error)
+type Repository interface {
+	CheckJobIsAvailable(ctx context.Context, jobName string) (bool, error)
+	Save(ctx context.Context, jobID uint64, executionContext *clientCPproto.RequestForExecute) error
+	Delete(ctx context.Context, key string) error
+	FetchNextJob(ctx context.Context) (string, *clientCPproto.RequestForExecute, error)
 }
-
 type jobRepository struct {
 	etcdClient etcd.Client
 }
 
-//NewJobExecutionRepository initializes jobExecutionRepository with the given etcdClient and scheduler
-func NewJobRepository(client etcd.Client) JobRepository {
+const (
+	pendingPrefix = "jobs/pending/"
+)
+
+// NewJobRepository initializes jobRepository with the given etcdClient
+func NewJobRepository(client etcd.Client) Repository {
 	return &jobRepository{
 		etcdClient: client,
 	}
 }
 
-func storeEnvVariablesInDatabase(ctx context.Context, etcdClient etcd.Client, jobId string, jobData map[string]string) error {
-
-	for envName, value := range jobData {
-		key := "jobs/" + jobId + "/env/" + envName
-		err := etcdClient.PutValue(ctx, key, value)
-		if err != nil {
-			return err
+// CheckJobIsAvailable returns true if given job is available otherwise returns false
+func (j jobRepository) CheckJobIsAvailable(ctx context.Context, jobName string) (bool, error) {
+	_, err := j.etcdClient.GetValue(ctx, "metadata/"+jobName)
+	if err != nil {
+		if err.Error() == constant.NoValueFound {
+			return false, errors.New("job with given name not found")
+		} else {
+			return false, err
 		}
 	}
-	return nil
+	return true, nil
 }
 
-func (j jobRepository) ExecuteJob(ctx context.Context, jobIdString string, jobName string, jobData map[string]string) error {
-	key := "jobs/" + jobIdString + "/metadataKeyName"
-	value := "metadata/" + jobName
-	err := j.etcdClient.PutValue(ctx, key, value)
+// Save takes jobID and executionContext and save it in database as pendingList
+func (j jobRepository) Save(ctx context.Context, jobID uint64, executionContext *clientCPproto.RequestForExecute) error {
+	jobIDasString := strconv.FormatUint(jobID, 10)
```

Comment:
this is not required because if you want to use it as a string in the key variable you can directly format there.